src/generator/generate-code.ts

```diff
@@ -223,9 +235,24 @@ export default async function generateCode(
 
   if (dmmfDocument.shouldGenerateBlock("inputs")) {
     log("Generating input types...");
-    dmmfDocument.schema.inputTypes.forEach(type =>
-      generateInputTypeClassFromType(project, resolversDirPath, type, options),
+    const inputTypes = dmmfDocument.schema.inputTypes;
+    const batchSize = Math.max(5, Math.ceil(inputTypes.length / (cpus().length * 2)));
+    
+    const batches = Array.from(
+      { length: Math.ceil(inputTypes.length / batchSize) },
+      (_, i) => inputTypes.slice(i * batchSize, (i + 1) * batchSize)
     );
+
+    for (const batch of batches) {
+      await Promise.all(
+        batch.map(async type => {
+          if (cacheManager.shouldRegenerateInput(type)) {
+            await generateInputTypeClassFromType(project, resolversDirPath, type, options);
+            cacheManager.updateInputCache(type);
+          }
+        })
+      );
+    }
```

Comment:
I'm not sure how this can speed up things.
`generateInputTypeClassFromType` is not an async function, returning a Promise or using callbacks.
Wrapping it into `Promise.all` won't alter the execution, because the event loop will be stuck with processing sync code of single iteration, and only when it finishes, it will go to the second one.
We would need a node threads to process things in parallel, on all the cpu's cores.